File: Leetcode/two_sum.py
# Method-1 (brute force):
class Solution:
    def twoSum(self, nums, target: int):
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
# Method-2:
class Solution:
    def twoSum(self, nums, target: int):
        dict = {}
        for i in range(len(nums)):
            num = target - nums[i]
            if num in dict:
                return [dict[num], i]
            dict[nums[i]] = i

File: LeetCode/TwoSum.cs
// Method-1 (Brute Force):
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        for (int i = 0; i < nums.Length - 1; ++i)
        {
            for (int j = i + 1; j < nums.Length; ++j)
            {
                if (nums[i] + nums[j] == target) return new int[] { i, j };
            }
        }
        return new int[] {};
    }
}

// Method-2 (Optimized):
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> dictNums = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; ++i)
        {
            int num = target - nums[i];
            if (dictNums.ContainsKey(num))
            {
                return new int[] { dictNums[num], i };
            }
            dictNums[nums[i]] = i;
        }
        return new int[] { };
    }
}
